app/models/solidus_paypal_braintree/transaction_address.rb

```diff
@@ -44,15 +44,25 @@ def spree_state
 
     def to_spree_address
       address = ::Spree::Address.new(
-        first_name: first_name,
-        last_name: last_name,
         city: city,
         country: spree_country,
         address1: address_line_1,
         address2: address_line_2,
         zipcode: zip
       )
 
+      if !first_name.nil?
+        ::Spree::Deprecation.warn("first_name and last_name are deprecated. Use name instead.", caller)
+        address.first_name = first_name
+        address.last_name = last_name || "(left blank)"
+      elsif address.respond_to? :name
+        address.name = name
+      else
+        first_name, last_name = split_name(name)
```

Comment:
Don't we have something in Solidus Core that does the same thing? I recall we have a class for that

Reply:
In Solidus 2.11, there's a `Spree::Address::Name` virtual attribute which handles splitting, it's what line 58-59 handle. I'm not aware of any class that handles name splitting before 2.11 - at least, I took a quick look just now and couldn't find one!

Reply:
I think we should use that class if it's defined. What do you think?

Reply:
`address.name` creates a new instance of `Spree::Address::Name` and sets the first and last names accordingly: https://github.com/solidusio/solidus/blob/44da6ff1061ac8dfd339f02228bbd921b4a4c006/core/app/models/spree/address.rb#L204-L210 - so above, if that class is available, we're using it as the name splitter.

Reply:
Oh, you are 100% right, thanks Sean!